include/mppic/motion_models.hpp

```diff
@@ -0,0 +1,61 @@
+#ifndef MPPIC__OPTIMIZATION__MOTION_MODELS_HPP_
+#define MPPIC__OPTIMIZATION__MOTION_MODELS_HPP_
+
+#include <cstdint>
+#include <string>
+#include <unordered_map>
+
+namespace mppi
+{
+
+class MotionModel
+{
+public:
+  MotionModel() = default;
+  ~MotionModel() = default;
+
+  /**
+   * @brief Predict velocities for given trajectories the next time step
+   *
+   * @param state for given time_step, tensor of shape
+   * [batch_size, ...] where last dim could be 5 or 7 depending on motion model used
+   *
+   * @return predicted velocities of the robot: tensor of shape [batch_size, ... ]
+   * where last dim could be 2 or 3 depending on motion model used
+   */
+  virtual xt::xtensor<double, 2> predict(
+    const xt::xtensor<double, 2> & state, const std::array<uint8_t, 2> & idx)
```

Comment:
We passing here whole state for one moment of a time. We need here StateIdxs instead of std::array

Reply:
If you check the useage, I updated it so that it uses only the indices: `idx[0], idx[1]`. Before, this was doing `idx.cbegin(), idx.cend()` which did require then `idx` to be the `StateIdxs`. But now we're just using the array values since this function is passed `state.idx.getControlRange()` rather than `state.idx` 

This was required so that we could decouple the `State` class from the `MotionModel` class so that there wasn't a circular dependency

Reply:
Essentially, I changed the pass to just the control ranges (`crange_` in prior code) which was the only thing from `StateIdxs` being accessed. This makes now `motion_models.hpp` now no longer depend on `state.hpp`. State depends on motion models so it can't go both ways. 

The signature was changed and the actual passed value was updated appropriately as well

Reply:
OK, I have an idea, give me like 15 minutes

Reply:
OK take a look